nistats/tests/test_datasets.py

```diff
@@ -171,26 +167,26 @@ def _expected_output_data_from_test_file():
         file_data.columns = ['trial_type', 'onset', 'duration']
         return file_data
     
+    @within_temp_dir
     def run_test():
         data_for_tests = _input_data_for_test_file()
         expected_data_from_test_file = _expected_output_data_from_test_file()
-        with NamedTemporaryFile(mode='w',
-                                dir=os.getcwd(),
-                                suffix='.csv') as temp_csv_obj:
-            data_for_tests.to_csv(temp_csv_obj.name,
```

Comment:
Using `.name` attribute is not needed since `to_csv` function also works with file objects (e.g temp_csv_obj). This will avoid a reopening of the same file and probably fix the related windows issue (multiple file access are forbidden).